# Hometask 5

### Задача 1

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр — 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверьте эту гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

Решение:

In [1]:
d = 17 # M(X)
alpha = 0.025 # a
n = 100 # n
d_mid = 17.5 # M(X) 
sigma = 4 # D(X)

Гипотеза $H_0$: $M(X) = 17$

Гипотеза $H_1$: $M(X) = 17.5$

Уровень значимости $\alpha = 0.025$ при правосторонней критической области $\Omega_{\alpha} = (t_{1 - \alpha}, \infty)$

Центральная предельная теорема:

$$a = M(X), \text{ } \sigma = \sqrt {\frac{D(X)}{n}}$$

Т.к. распределение нормальное, то можно применить правло двух сигм (95%)

$$\left( 17 - 2 \cdot \sqrt{\frac{D(X)}{n}}, 17 + 2 \cdot \sqrt{\frac{D(X)}{n}} \right)$$

In [5]:
d + 2 * (sigma / n) ** 0.5

17.4

Правосторонняя критическая область $( 17.4, \infty )$

Средний диаметр 17,5 попадает в критическую область. Значит верна гипотеза $H_1$

### Задача 2

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением — 16. Найдите доверительный интервал для оценки математического ожидания с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

Решение:

In [14]:
std = 16
reliability = 0.95
n = 256
m = 80

Квантиль t-распределения $(t_{1 - \alpha/2, \: n - 1})$

In [15]:
from scipy import stats

In [19]:
# n - 1
t_n1 = stats.t(n - 1)
# 1 - alpha / 2
t = t_n1.ppf(1 - (1 - reliability) / 2)
t

1.9693105698491928

$$P \left( \overline{X} - t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} \leq M(X) \leq \overline{X} + t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} \right) = 1 - \alpha$$

In [21]:
(m - t * std / n ** 0.5, m + t * std / n ** 0.5)

(78.0306894301508, 81.9693105698492)

### Задача 3

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

Решение:

In [23]:
import numpy as np

In [24]:
n = 10
alpha = 0.01
weight = 200
data = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

$$P \left( \overline{X} - t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} \leq M(X) \leq \overline{X} + t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} \right) = 1 - \alpha$$

In [26]:
mean = data.mean()
mean

198.5

In [27]:
std = data.std(ddof=1)
std

4.453463071962462

In [28]:
# n - 1
t_n1 = stats.t(n - 1)
# 1 - alpha / 2
t = t_n1.ppf(1 - alpha / 2)
t

3.2498355440153697

In [30]:
(mean - t * std / n ** 0.5, mean + t * std / n ** 0.5)

(193.92322840036354, 203.07677159963646)

Средний вес попадает в доверительный интервал, значит утверждение продавца верно.